lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -585,7 +585,10 @@ def get_device_facts(self):
 
             d['partitions'] = {}
             for folder in os.listdir(sysdir):
-                m = re.search("(" + diskname + r"\d+)", folder)
+                if "nvme" or "cciss" or "loop" in diskname:
```

Comment:
Wouldn't it be more correct to verify if diskname start by those strings ?

Reply:
May be. This is "backport" from https://github.com/ceph/ceph-ansible/pull/2219. Regex was replaced by a match in string.

Reply:
I agree, startswith is less likely to collide with other device names.